<launch>
    <arg name="circle_offsets" default="[0.1, 0.3]" />

    <arg name="lattice_num_layers" default="8" />
    <arg name="lattice_num_lateral_samples_per_side" default="8" />
    <arg name="lattice_layer_spacing" default="2.0" />
    <arg name="lattice_lateral_spacing" default="0.05" />
    <arg name="lattice_k_length" default="0.5" />

    <arg name="ref_max_speed" default="5.0" />
    <arg name="ref_max_lat_acc" default="1.0" />
    <arg name="ref_max_lon_acc" default="1.0" />
    <arg name="ref_max_lon_dec" default="1.0" />

    <arg name="track_num_paths" default="9" />
    <arg name="wheelbase" default="0.3" />
    <arg name="max_steering_angle" default="0.4" />
    <arg name="track_path_lateral_spacing" default="0.06" />
    <arg name="track_look_ahead_time" default="0.5" />

    <arg name="global_path_topic" default="global_path" />
    <arg name="odom_topic" default="odom" />
    <arg name="drive_feedback_topic" default="drive_feedback" />

    <node name="local_planner" pkg="local_planner" type="local_planner_node" output="screen">
        <rosparam param="circle_offsets" subst_value="True">$(arg circle_offsets)</rosparam>

        <param name="lattice_num_layers" value="$(arg lattice_num_layers)" />
        <param name="lattice_num_lateral_samples_per_side" value="$(arg lattice_num_lateral_samples_per_side)" />
        <param name="lattice_layer_spacing" value="$(arg lattice_layer_spacing)" />
        <param name="lattice_lateral_spacing" value="$(arg lattice_lateral_spacing)" />
        <param name="lattice_k_length" value="$(arg lattice_k_length)" />

        <param name="ref_max_speed" value="$(arg ref_max_speed)" />
        <param name="ref_max_lat_acc" value="$(arg ref_max_lat_acc)" />
        <param name="ref_max_lon_acc" value="$(arg ref_max_lon_acc)" />
        <param name="ref_max_lon_dec" value="$(arg ref_max_lon_dec)" />

        <param name="track_num_paths" value="$(arg track_num_paths)" />
        <param name="wheelbase" value="$(arg wheelbase)" />
        <param name="max_steering_angle" value="$(arg max_steering_angle)" />
        <param name="track_path_lateral_spacing" value="$(arg track_path_lateral_spacing)" />
        <param name="track_look_ahead_time" value="$(arg track_look_ahead_time)" />

        <remap from="global_path" to="$(arg global_path_topic)"/>
        <remap from="odom" to="$(arg odom_topic)"/>
        <remap from="drive_feedback" to="$(arg drive_feedback_topic)"/>
    </node>
</launch>
